arch/arm/core/aarch64/reset.S

```diff
@@ -19,6 +19,15 @@
 
 _ASM_FILE_PROLOGUE
 
+/*
+ * Platform may do platform specific init at EL3.
+ * The function implementation must preserve callee saved registers as per
+ * Aarch64 ABI PCS.
+ */
+WTEXT(z_arch_el3_plat_init)
+SECTION_FUNC(TEXT,z_arch_el3_plat_init)
+ret
```

Comment:
Most EL3 settings should be common between Armv8-A SoCs as done below.. Why is this needed?

Reply:
As I mentioned in PR description currently GICv3 init is one thing that is qualifying as platform specific. Also few ACTLR init must be done as soon as core is reset as per arm recommendation. Those are indeed platform specific. And some of them are core specific. This is only based on v8.0 analysis.  Beyond that even more such plat_specific inits are foreseen.